### ISSUE TITLE
Remove isInMemory mode

### ISSUE THREAD

**Issue by semihsalihoglu-uw** [kuzudb/kuzu#1222]
This is another feature we should remove. I have stopped finding this very meaningful. In memory mode currently forces database files to be mapped in buffer manager but it's not very clear to me if there is benefit to this, especially given that we're moving to an mmap dynamicly growing bm. If there is enough BM space, we should just be able to map db files to memory anyways. 

What we may eventually need is a feature for temporary databases, which shouldn't be mapped to disk files if we implement features that allow directly scanning node and rel records from sources with the intention of querying only. We could support a readOnly temporary db mode, but it's too early to have this discussion. 

In short, this is currently slowing us down, so let's remove this. 

Note: I didn't do the work of thinking about what it would take to remove this (changing some tests might take a bit of time). But regardless, before the next we should at least turn it off, so no one can define an in memory db. And don't forget to add this to the release notes as well. Even though it's minor, let's be complete.

**Comment by acquamarin** [comment]
solved in #1224
